web/app/services/screensaver.service.js

```diff
@@ -0,0 +1,349 @@
+            $location.url("/view/" + nextDbId);
+        }
+
+        var slideshow = function () {
+            log("Screensaver (" + _config.onStart.type + ") started in dashboard " + $route.current.params.i);
+            _currentDbIndex = 0;
+            nextDashboard();
+            _slideshowTimer = $interval(nextDashboard, (_config.slideshowIntervalSec || 10) * 1000);
+        }
+
+        var start = function () {
+            if (_isRunning)
+                return;
+
+            _isRunning = true;
+            watchEvents();
+
+            if (_config.onStart.type === 'slideshow') {
+                slideshow();
+            } else {
+                $location.url("/view/" + _config.onStart.dashboard)
+            }
+        };
+
+        var stop = function (isFromOHService) {
+            if (!_isRunning)
+                return;
+
+            unWatchEvents();
+            $interval.cancel(_slideshowTimer);
+            _slideshowTimer = null;
+            _isRunning = false;
+            log("Screensaver stopped.");
+
+            if (_config.isEnabled) {
+                idleTimerStart();
+            }
+
+            if (isFromOHService)
+                return;
+
+            if (_config.onStop.type === 'gotodashboard') {
+                $location.url("/view/" + _config.onStop.dashboard);
+            }
+        };
+
+        var idleTimerStart = function () {
+            _idleTimer = $timeout(onIdle, _config.idleTimeoutSec * 1000);
+        }
+
+        var idleTimerStop = function () {
+            $timeout.cancel(_idleTimer);
+            _idleTimer = null;
+        }
+
+        var init = function () {
+            $timeout(function () {
+                initConfig();
+                if (!_config || !_config.idleTimeoutSec || !_config.isEnabled)
+                    return;
+                idleTimerStart();
+            });
+        };
+
+        var saveSettings = function (config) {
+            config = config || _config;
+            // Uniqify our arrays
+            function onlyUnique(value, index, self) { return self.indexOf(value) === index; }
+            config.onStart.dashboards = config.onStart.dashboards.filter(onlyUnique);
+            config.onStart.dashboardsExcluded = config.onStart.dashboardsExcluded.filter(onlyUnique);
+            $rootScope.settings.screensaver = _config;
+            OH2StorageService.saveCurrentPanelConfig().then(function () {
+                _config = config;
+                init();
+            });
+            return true;
+        }
+
+        Object.defineProperty(this, "isEnabled", {
+            get: function () { return _config.isEnabled; },
+            set: function (v) {
+                _config.isEnabled = v;
+                saveSettings(_config);
+            }
+        })
+
+        Object.defineProperty(this, "config", {
+            get: function () { return _config; }
+        })
+
+        var reConfig = function () {
```

Comment:
Maybe this isn't needed anymore now that the dashboard rotation is stored in the panel config? The only remaining check necessary as I see it is if a dashboard has been removed while it's part of the screensaver rotation.